tests/ContainsManyTest.php

```diff
@@ -35,13 +36,13 @@ protected function init(): void
         $this->addField('amount', ['type' => 'money']);
 
         // will contain many Lines
-        $this->containsMany('lines', ['model' => [Line2::class], 'caption' => 'My Invoice Lines']);
+        $this->containsMany($this->fieldName()->lines, ['model' => [Line2::class], 'caption' => 'My Invoice Lines']);
 
         // total_gross - calculated by php callback not by SQL expression
         $this->addCalculatedField($this->fieldName()->total_gross, function (self $m) {
             $total = 0;
-            foreach ($m->ref('lines') as $line) {
-                $total += $line->get('total_gross');
+            foreach ($m->lines as $line) {
```

Comment:
this is one strong advantage of hintable magic, thanks to https://github.com/atk4/data/pull/821 foreach value has correct type for phpstan as the iterated model